### ISSUE TITLE
Why does a catch leave the outer loop?

### ISSUE THREAD

**Issue by reks99** [fesch/Structorizer.Desktop#819]
Hi,

I wrote the following program to test the try-catch-final element. 
![Taschenrechner](https://user-images.githubusercontent.com/18035140/74940765-371c1600-53f2-11ea-870c-5c3a86948fd5.png)
The called function looks like
![rechne-3](https://user-images.githubusercontent.com/18035140/74941573-8adb2f00-53f3-11ea-943f-0102466566ff.png)

The try-catch is embedded in an endless loop, so in my opinion the error is catched and served in the catch path. After that the loop should be continued - except there is an explicit exit in the catch path.

Fine regards
Rolf.

**Comment by codemanyak** [comment]
The problem is the empty instruction in the *finally* section you added in order to place a comment. This causes a new error after having caught and handled the `fehler`.
Traditionally empty instruction elements (which is not the same as an empty sequence not consisting of any element and being marked with the empty set symbol looking like Ø) and empty lines in multi-line instruction elements cause execution errors.

**Comment by codemanyak** [comment]
I found a real bug: the exit event is unduly caught by the try block.

**Comment by reks99** [comment]
Hi Kay,

> The problem is the empty instruction in the finally section you added in order to place a comment.

Why alters a comment the command area? Again, comments should be comments, I think.

Fine regards
Rolf

**Comment by codemanyak** [comment]
A comment is a comment. But executor detests empty instruction parts. This is the problem. Of course this may seem somewhat arbitrary - why not allow noop instructions? I can only guess here - maybe because of the following reasons:
1. in order to avoid messing a diagram with empty elements
1. to alert users on debugging a diagram that was initially designed via comments (e.g. in switch comment - text mode) and hasn't been completely filled with real instructions then.
1. to discourage huge multiline instruction elements grouped by empty lines (instead of using several instruction elements).

This is of course inconvenient in case you want to place a comment into an otherwise empty branch of an alternative or TRY block...

**Comment by codemanyak** [comment]
What we perhaps might do is just popping up a warning (or write it to the output window if enabled) on detecting an empty instruction element or line without aborting the execution. This is a lesser hassle.

**Comment by codemanyak** [comment]
A deeper analysis show that it might indeed be a bug. The strange message `<> is not a correct function` is a consequence that executor just falls back to try a built-in function if all other explicit variants fail, among which there are assignments, Turtleizer commands, declarations, and type definitions. So possibly we just skip it without warning and without incrementing the execution counter...

**Comment by codemanyak** [comment]
@reks99 
I offer you an updated alpha version (alpha1) for 3.30-06.
[Structorizer_3.30-06alpha1.zip](https://github.com/fesch/Structorizer.Desktop/files/4233549/Structorizer_3.30-06alpha1.zip)

**Comment by reks99** [comment]
@codemanyak 
Hi Kay,

I tested the alpha version with the try-demo from the documentation. The debugger died at the first step. The current stable version works.

Fine regards
Rolf

**Comment by codemanyak** [comment]
@reks99 
Sorry, I can't reproduce what you wrote. Neither does my Executor die nor does an error from the TRY block leave the loop. (???)

**Comment by codemanyak** [comment]
Yet I detected that Analyser ignores the interior of TRY elements, which is a bug.

**Comment by reks99** [comment]
Hi Kay,

I put my ini-file - a fresch one - and the errors when executing the try-catch example in the attached file. Hope that helps

[error.zip](https://github.com/fesch/Structorizer.Desktop/files/4236498/error.zip)

Fine regards
Rolf

**Comment by codemanyak** [comment]
@reks99 
This helped indeed. I had accidently smuggled in a Java 11 method. Sorry for the trouble.
Here is an improved and cleaned alpha 2 version, containing fixes upto #825:
[Structorizer_3.30-06alpha2.zip](https://github.com/fesch/Structorizer.Desktop/files/4236716/Structorizer_3.30-06alpha2.zip)

**Comment by reks99** [comment]
Hi Kay,

I can report a partial success. The try-demo works well, but my program only worked once. I add the Program and the error output.

[test.zip](https://github.com/fesch/Structorizer.Desktop/files/4237114/test.zip)

Hope that help
Rolf

**Comment by codemanyak** [comment]
This time I am lost. I ran the alpha2 jar under Windows, under Linux (Debian), starting with Arranger or Structorizer, all without trouble, repeatedly. The stacktrace in try.error tells a rather impossible story. There is hardly anything within method Executor.initInterpreter() that could be null.

**Comment by reks99** [comment]
Hi Kay,

First of all until now the Taschenrechner runs in the debugger. But it was funny, after read your last comment I tested twice again and get the exception I send you. Then I unpacked the zip file I send you and tested the program and surpries I runs flawlessly. Then I made a diff on the unziped and the working files and no difference is reported. Now I run my working files agai and get no error even a few minutes again.

So I think the problems have been fixed.
Fine regards
Rolf
